chess-app/src/main/java/lt/codeacademy/vigi40/chessapp/converters/ChessPlayerConverter.java

```diff
@@ -0,0 +1,268 @@
+                throw new IllegalArgumentException("" +
+                        "Surname must be at least 2 characters long and contain only letters and hyphen");
+            }
+            if (addChessPlayerDTO.getEmail() != null && !addChessPlayerDTO.getEmail().
+                    matches("^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$")) {
+                throw new IllegalArgumentException("Invalid email format");
+            }
+            if (addChessPlayerDTO.getPersonCode() != null
+                    && !isValidPersonCode(addChessPlayerDTO.getPersonCode().toString())) {
+                throw new IllegalArgumentException("Person code should meet LT requirements");
+            }
+            if (addChessPlayerDTO.getStartPlayChessFromDate() != null
+                    && addChessPlayerDTO.getStartPlayChessFromDate().compareTo(LocalDate.now()) > 0) {
+                throw new IllegalArgumentException("Start play chess from date cannot be empty or upcoming date");
+            }
+            chessPlayer = new ChessPlayer();
+            chessPlayer.setName(addChessPlayerDTO.getName() != null
+                    ? addChessPlayerDTO.getName().toUpperCase() : null);
+            chessPlayer.setSurname(addChessPlayerDTO.getSurname() != null
+                    ? addChessPlayerDTO.getSurname().toUpperCase() : null);
+            chessPlayer.setEmail(addChessPlayerDTO.getEmail() != null
+                    ? addChessPlayerDTO.getEmail().toLowerCase() : null);
+            chessPlayer.setPersonCode(addChessPlayerDTO.getPersonCode());
+            chessPlayer.setStartPlayChessFromDate(addChessPlayerDTO.getStartPlayChessFromDate());
+        }
+        return chessPlayer;
+    }
+
+    private static String getSexFromPersonCode(Long personCode) {
+        int firstDigit = (int) (personCode / 10000000000L); // extract first digit
+        if (firstDigit == 3 || firstDigit == 5) {
+            return "male";
+        } else if (firstDigit == 4 || firstDigit == 6) {
+            return "female";
+        } else {
+            throw new IllegalArgumentException("Invalid person code");
+        }
+    }
+
+    private static int getAgeFromPersonCode(Long personCode) {
+        int year;
+        int firstDigit = (int) (personCode / 10000000000L); // extract first digit
+        int secondAndThirdDigits = (int) (personCode / 100000000L) % 100; // extract second and third digits
+        if (firstDigit == 3 || firstDigit == 4) {
+            year = 1900 + secondAndThirdDigits;
+        } else if ((firstDigit == 5 || firstDigit == 6) && secondAndThirdDigits <= LocalDate.now().getYear() % 100) {
+            year = 2000 + secondAndThirdDigits;
+        } else {
+            throw new IllegalArgumentException("Invalid person code");
+        }
+        int month = (int) (personCode / 1000000L) % 100;
+        int day = (int) (personCode / 10000L) % 100;
+        if ((month > 0 && month < 13) && (day > 0 && day < 32)) {
```

Comment:
Paprastesniam kodo skaitymui / prieziurai ir norint sumazint kodo nestinima ifuose - vietoj to, kad turet ilga salyga, tokia kaip `((month > 0 && month < 13) && (day > 0 && day < 32))`, joje daug logikos ir galiausiai paprasta else salyga - siulyciau apversti salyga, ir padaryti taip:

```
if(!(month > 0 && month < 13) || !(day > 0 && day < 32)){
            throw new IllegalArgumentException("Invalid person code");
}

...
visa likusi logika, bet nestinta jau vienu lygiu maziau
...
```